src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObject.cs

```diff
@@ -29,6 +29,9 @@ public CryptographicAttributeObject(Oid oid, AsnEncodedDataCollection? values)
             {
                 foreach (AsnEncodedData asn in values)
                 {
+                    if (asn.Oid is null)
+                        throw new ArgumentException(SR.Argument_InvalidOidValue, nameof(values));
+
                     if (!string.Equals(asn.Oid!.Value, oid.Value, StringComparison.Ordinal))
```

Comment:
I left the null forgiving operator in here as I wasn't sure if it should be removed (since `asn.Oid` should never be null once it reaches here.